streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -727,6 +727,23 @@ object ZTransducer {
       }
     }
 
+  /**
+   * Creates a transducer that limits byte chunks to the given size.
+   */
+  def throttleBytes(limit: Int): ZTransducer[Any, Nothing, Byte, Chunk[Byte]] =
+    throttleChunks[Byte](limit)
+
+  /**
+   * Creates a transducer that limits chunks to the given size.
+   */
+  def throttleChunks[A](limit: Int): ZTransducer[Any, Nothing, A, Chunk[A]] = {
+    @scala.annotation.tailrec
+    def go(z: Chunk[Chunk[A]], chunk: Chunk[A]): Chunk[Chunk[A]] =
+      if (chunk.isEmpty) z else go(z + chunk.take(limit), chunk.drop(limit))
+
+    ZTransducer.identity[A].mapChunks(go(Chunk.empty, _))
+  }
```

Comment:
Let's move this to a combinator on `ZStream` so it doesn't have the chunks exposed. Combinators that alter the chunking structure are not a great fit for transducers.

Also, we need a better name: `chunkUpTo`, `chunkMaxN`, something like that.

Reply:
Consider an endpoint defined as
```scala
type Endpoint[-R, +E] = ZTransducer[R, E, Byte, Int]

def endpoint[R, E](limit: Int): Endpoint[R, E] =
  throttleChunks[Byte](limit).mapM(write)

def write(chunk: Chunk[Byte]): UIO[Int] = ???
```
Moving the combinator to `ZStream` would require enforcing the limit upstream.
I agree that exposing the chunks is not ideal but that is unavoidable in this case.
Perhaps we could make an exception? 

Is the name `limitChunks` okay?

Reply:
`chunkLimit` would be great; the idea is to be completion friendly in the sense that all chunking-related methods can be found by typing `stream.chunk...`.

> Moving the combinator to ZStream would require enforcing the limit upstream. I agree that exposing the chunks is not ideal but that is unavoidable in this case.
Perhaps we could make an exception?

It's probably going to end up looking pretty similar to your current code, so give it a try and let me know if you need a hand.

Reply:
This is how the stream combinator would look:
```scala
def chunkLimit(max: Int): ZStream[R, E, Chunk[O]] =
  mapChunks(Chunk.single).mapConcatChunk {chunk =>
    @scala.annotation.tailrec
    def go(z: Chunk[Chunk[O]], chunk: Chunk[O]): Chunk[Chunk[O]] =
      if (chunk.isEmpty) z else go(z + chunk.take(max), chunk.drop(max))
    go(Chunk.empty, chunk)
  }
```
Do you want the combinator in `ZTransducer` to be removed?

Reply:
Or, the combinator in `ZTransducer` could be defined as:
```scala
def chunkLimit[A](max: Int): ZTransducer[Any, Nothing, Chunk[A], Chunk[A]]
```

Reply:
The combinator has been revised. Please review.